app/controllers/admin/orders_controller.rb

```diff
@@ -1,10 +1,59 @@
 class Admin::OrdersController < Admin::AdminController
   before_action :load_order_info, only: :index
+  before_action :load_order, except: :index
+  before_action :order_params, only: :update
+
   def index
     @pagy, @orders = pagy Order.all.order_newest
   end
 
+  def update
+    result = @order.update(status: order_params["status"])
+    unless result
+      flash.now["danger"] = t "update_order_fail"
+      render :index
+    end
+    handle_update_quantity_book result
+    send_mail_notification
+    flash["success"] = t "update_order_success"
+    redirect_to admin_orders_path
+  end
+
+  private
+  def order_params
+    params.require(:order).permit(:status, :id)
+  end
+
+  def load_order
+    @order = Order.find_by id: params[:id]
+    return if @order
+
+    flash[:danger] = t "not_found"
+    redirect_to admin_order_path
+  end
+
   def load_order_info
     @book_details = BookDetail.all
   end
+
+  def handle_update_quantity_book result
+    return unless result && order_params["status"] == "accepted"
+
+    @order.order_details.each do |order_detail|
```

Comment:
Đưa vào trong transaction đi em

Reply:
Hàm này em cũng cần đưa vào transaction nhé, nếu chỉ cẩn 1 bản ghi trong này không update đc thì phải rollback lại toàn bộ nhé